lib/rmg/WaterProxy.cpp

```diff
@@ -142,6 +142,13 @@ RouteInfo WaterProxy::waterRoute(Zone & dst)
 				dst.areaPossible().subtract(lake.neighbourZones[dst.getId()]);
 				continue;
 			}
+
+			//Don't place shipyard or boats on the very small lake
```

Comment:
Will it try to put any treasures there?

Reply:
Yep, but it's irrelevant to the gameplay.

Such small lakes are usually placed right next to land guard between the zones.